docs/cep/query-guide/query.md

```diff
@@ -40,9 +40,132 @@ The following parameters are used to configure a stream definition.
-| `grouping clauses` | `GROUP BY` functions to group and organize output.  |
+| `grouping clauses` | `GROUP BY` functions to group and organize output. For more information, refer to [GROUP BY](group-by).  |
+
+## Query Projections
+
+Query projection allow you to filter and transform streaming data by choosing which fields or attributes of a data stream you want to include or exclude in query results. By selecting only the necessary fields and performing any necessary calculations or aggregations, query projection can help to reduce the amount of data that needs to be processed and improve the overall performance of streaming applications.
+
+In Macrometa stream worker queries, projection is specified using the `SELECT` clause. This clause is used to select a subset of fields from the incoming data stream and return them in the query results. The `SELECT` clause can also be used to perform transformations on the selected fields, such as aggregations or calculations.
+
+Here are some examples things you can do with query projections in Macrometa stream workers.
+
+### Select All Fields or Specific Fields
+
+Select some or all fields from the input stream to be inserted into an output stream.
+
+For example, to select only the `name` and `age` fields from a stream of customer data, you could use the following query:
+
+```sql
+INSERT INTO OutputStream
+SELECT name, age
+FROM customers;
+```
+
+You can select all attributes in an input stream by using an asterisk (*) wildcard or by omitting the `SELECT` statement.
+
+```sql
+INSERT INTO OutputStream
+SELECT *
+FROM InputStream;
+```
+
+Or use:
+
+```sql
+INSERT INTO OutputStream
+FROM InputStream;
+```
+
+### Rename Attributes
+
+Select attributes from an input stream and insert them into the output stream with different names.
+
+For example, this query renames `roomNo` to `roomNumber` and `temp` to `temperature`.
+
+```sql
+INSERT INTO OutputStream
+SELECT roomNo AS roomNumber, temp AS temperature
+FROM InputStream;
+```
+
+### Introduce Constant Values
+
+You can add constant values by assigning them to an attribute using `as`.
+
+For example, this query specifies `C` to be used as the constant value for `scale` attribute.
+
+```sql
+INSERT INTO OutputStream
+SELECT roomNo, temp, 'C' AS scale
+FROM InputStream;
+```
+
+### Filter by Field Value
+
+You can use the `WHERE` or `HAVING` clauses to filter the incoming data stream based on a specific field value.
+
+For example, to select only the customer data where the `age` field is greater than 30, you could use the following query:
+
+```sql
+INSERT INTO OutputStream
+SELECT * 
+FROM InputStream 
+WHERE age > 30;
+```
+
+For more information, refer to [HAVING | WHERE](having-where.md).
+
+### Use Expressions
+
+This uses attributes with mathematical and logical expressions in the precedence order given below, and assigns them to the output attribute using `AS`.
```

Comment:
Probably fine, but I stumble a bit reading "This uses attributes" - maybe "Expressions use attributes with..."?